builtin/providers/azurerm/resource_arm_virtual_network.go

```diff
@@ -268,6 +293,40 @@ func resourceAzureSubnetHash(v interface{}) int {
 	}
 	return hashcode.String(subnet)
 }
+func getExistingSubnet(resGroup string, vnetName string, subnetName string, meta interface{}) network.Subnet {
```

Comment:
In order to raise an error, can we update this to return a `(*network.Subnet, error)`?